packages/cms-editor/src/main/control/AddContentObject.tsx

```diff
@@ -0,0 +1,149 @@
+import type { CmsData, ContentObject, MapStringString } from '@axonivy/cms-editor-protocol';
+import {
+  BasicField,
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  Flex,
+  hotkeyText,
+  Input,
+  selectRow,
+  Textarea,
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+  useHotkeys
+} from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useQueryClient } from '@tanstack/react-query';
+import { type Table } from '@tanstack/react-table';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useAppContext } from '../../context/AppContext';
+import { useClient } from '../../protocol/ClientContextProvider';
+import { useMeta } from '../../protocol/use-meta';
+import { genQueryKey, useQueryKeys } from '../../query/query-client';
+import { useKnownHotkeys } from '../../utils/hotkeys';
+import { useClientLanguage } from '../../utils/use-client-language';
+
+type AddContentObjectProps = {
+  table: Table<ContentObject>;
+};
```

Comment:
If I understand this correct, you currently only use this table to select the next row.
Maybe it is easier to understand and use this component if you are more specific here. So instead of have the whole table object why not give a function to select the next row?

Reply:
Then it would also be relatively easy to test the whole component if wished.